contracts/Gate.sol

```diff
@@ -38,9 +50,117 @@ contract Gate is ProtocolVersioned {
      *  Storage
      */
     address public workers; //TODO: update type once workers contract and interfaces are added to the repo.
+
+    // stake requests
     mapping(address /*staker */ => StakeRequest) public stakeRequests;
 
-    function Gate() public { //TODO: complete as necessary
+    // bounty amount
+    uint256 private bounty;
+
+    // utility token UUID
+    bytes32 private uuid;
+
+    // OpenSTValue contract
+    OpenSTValue private openSTValue;
+
+    /*
+     *  Public functions
+     */
+    constructor(
+        address _workers,
+        uint256 _bounty,
+        bytes32 _uuid,
+        address _openSTValue)
+        public
+        ProtocolVersioned(_openSTValue)
+    {
+        require(_workers != address(0));
+        require(_openSTValue != address(0));
+        require(_uuid.length != 0);
+
+        // Should bounty with 0 amount be allowed ?
```

Comment:
For the purposes of limiting the PoC scope, these kinds of questions we will set aside. Our testing will focus on the complete paths and not validations, so no need for validation `require`s.

Reply:
sure!!!